Cabal/Distribution/Simple/Configure.hs

```diff
@@ -1602,9 +1608,7 @@ checkForeignDeps pkg lbi verbosity = do
                      ++ ["-I."]
                      ++ collectField PD.cppOptions
                      ++ collectField PD.ccOptions
-                     ++ [ "-I" ++ dir
-                        | dep <- deps
-                        , dir <- Installed.includeDirs dep ]
+                     ++ installedIncludeDirs
```

Comment:
Personally I'd keep it inline here rather than pulling it out. e.g.
```
++ [ "-I" ++ dir
   , dir <- ordNub [ dir | dep <- deps
                         , dir <- Installed.includeDirs dep ]
            -- dedupe include dirs of dependencies
            -- to prevent quadratic blow-up
   ]
```